styx-common/src/test/java/com/spotify/styx/util/WorkflowValidatorTest.java

```diff
@@ -247,6 +248,44 @@ public void shouldNotAllowComponentWithHash(String component) {
         contains("component id cannot contain #"));
   }
 
+  @Parameters({"sa@.abc.com", "sa#@abc.com"})
+  @Test
+  public void shouldRejectInvalidServiceAccount(String serviceAccount) {
+    WorkflowConfiguration configuration = WorkflowConfigurationBuilder.from(FULL_WORKFLOW_CONFIGURATION)
+            .serviceAccount(serviceAccount)
+            .build();
+
+    assertThat(sut.validateWorkflow(Workflow.create("test", configuration)),
+            contains("service account format is not valid"));
+  }
+
+  @Test
+  public void shouldRejectSpaceServiceAccountWithTailingSpace() {
+    String[] invalidServiceAccounts = {"sa@abc.com ", "sa@abc.com\n"};
```

Comment:
Please add a comment about the we needed to use a rudimentary for loop instead of a parametric tests because JUnitParams removes the trailing spaces